server.js

```diff
@@ -7,22 +7,70 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+
+app.get('/', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+
+  if (thoughts.length > 0) {
+    res.json(thoughts)
+  }
+  else {
+    res.status(404).json({ message: 'No thoughts yet' })
+  }
 })
 
+app.post('/', async (req, res) => {
+  const { message } = req.body
+  const thought = new Thought({ message })
+
+  try {
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  }
+  catch (err) {
+    res.status(400).json({ message: 'Could not save thought to Database', error: err.error })
+  }
+})
+
+app.post('/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params
+  //a thought with a specific id
+  const thought = await Thought.findById(thoughtId)
+
+  if (thought) {
+    //likes
+    thought.hearts++
+    thought.save()
```

Comment:
This could actually be done in one line:
`const thought = await Thought.findOneAndUpdate({ _id: req.params.id }, { $inc: { hearts: 1 } }, { new: true })`